pkgs/by-name/ss/ssh-openpgp-auth/package.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, rustPlatform
+, fetchFromGitea
+, pkg-config
+, just
+, rust-script
+, installShellFiles
+, bzip2
+, nettle
+, openssl
+, sqlite
+, stdenv
+, darwin
+, openssh
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ssh-openpgp-auth";
+  version = "0.2.2";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "wiktor";
+    repo = "ssh-openpgp-auth";
+    rev = "ssh-openpgp-auth/${version}";
+    hash = "sha256-5ew6jT6Zr54QYaWFQIGYXd8sqC3yHHZjPfoaCossm8o=";
+  };
+  buildAndTestSubdir = "ssh-openpgp-auth";
+
+  cargoHash = "sha256-/k/XAp7PHIJaJWf4Oa1JC1mMSR5pyeM4SSPCcr77cAg=";
+
+  nativeBuildInputs = [
+    pkg-config
+    rustPlatform.bindgenHook
+    just
+    rust-script
+    installShellFiles
+  ];
+  # Otherwise just's build, check and install phases take precedence over
+  # buildRustPackage's phases.
+  dontUseJustBuild = true;
+  dontUseJustCheck = true;
+  dontUseJustInstall = true;
+
+  postInstall = ''
+    export HOME=$(mktemp -d)
+    just generate manpages ${pname} $out/share/man/man1
+    just generate shell_completions ${pname} shell_completions
+    installShellCompletion --cmd ${pname} \
+      --bash shell_completions/${pname}.bash \
+      --fish shell_completions/${pname}.fish \
+      --zsh  shell_completions/_${pname}
```

Comment:
over-interpolation of pname is discouraged.

Reply:
You are right.. I inherited these lines from [upstream's `package.nix`](https://codeberg.org/wiktor/ssh-openpgp-auth/src/branch/main/package.nix) which is shared for both projects. I rewrote both expressions to use a `./generic.nix` expression, like upstream's `package.nix`. Since upstream is with good contact with me, I'm fairly sure they won't introduce too much deviation in the future between the requirements for the daemon and the client, so it should be a good approach for the near future.